index.js

```diff
@@ -10,47 +10,76 @@ module.exports = {
     dependencies: 'vision',
     multiple: true,
     register: (server, options) => {
-      const { model, basePath } = options
+      const { modelOptions, configs } = options
 
       let hasRootPage = false
+      const forms = {}
+      configs.forEach(config => {
+        forms[config.id] = new Model(config.configuration, {...modelOptions, basePath: config.id})
+      })
 
-      model.pages.forEach(page => {
-        // GET
-        let route = page.makeGetRoute(model.getState)
-        let path = route.path
-        if (path === '/') {
-          hasRootPage = true
-        }
-        if (route.path) {
-          path = path !== '/' ? path.replace(/^/, `/${basePath}`) : `/${basePath}`
+      server.route({
+        method: 'post',
+        path: `/publish`,
+        handler: (request, h) => {
+          const {id, configuration} = request.payload
+          let model = new Model(configuration, modelOptions)
+          forms[id] = model
+          return h.response({}).code(204)
         }
-        route.path = path
-        server.route(route)
+      })
 
-        // POST
-        if (page.hasFormComponents) {
-          let postPath = page.makePostRoute(model.mergeState)
-          postPath.path = path
-          server.route(postPath)
+      server.route({
+        method: 'get',
+        path: `/{id}/{path*}`,
+        handler: (request, h) => {
+          const { path, id } = request.params
+          let model = forms[id]
+          if (!model) {
+            console.log('oops')
+          }
+          let page = model.pages.find(page => page.path.replace(/^\//,'') === path)
+          if (page) {
+            return page.makeGetRouteHandler(model.getState)(request, h)
+          } else {
+            console.log('err')
+          }
         }
       })
 
-      if (!hasRootPage) {
-        server.route({
-          method: 'get',
-          path: `/${basePath}`,
+      let handleFiles =  (request, h) => {
+        let { uploadService } = request.services([])
+        return uploadService.handleUploadRequest(request, h)
+      }
+
+      server.route({
+        method: 'post',
+        path: `/{id}/{path*}`,
+        config: {
+          payload: {
+            output: 'stream',
+            parse: true,
+            maxBytes: Number.MAX_SAFE_INTEGER,
```

Comment:
Let's make this something like 5mb. Also we'll need to handle the error wheen file is too large and report back to UI properly.